hooks/post-checkout

```diff
@@ -34,7 +34,14 @@ merge() {
   export GIT_COMMITTER_EMAIL="foo@bar"
   export GIT_AUTHOR_NAME="github-merged-pr-buildkite-plugin"
   export GIT_COMMITTER_NAME="github-merged-pr-buildkite-plugin"
-  git merge "${BUILDKITE_COMMIT}"
+  if ! git merge "${BUILDKITE_COMMIT}"; then
```

Comment:
Is the logic that if `git merge` fails, i.e., with a nonzero return code—it'll then run `git merge --abort`?

This might not work, since the script has `-euo pipefail` set which causes the script to end on any nonzero return code.

Reply:
Bash/sh won't exit when the failure occurs in an `if`, since that's explicitly handling a failure. E.g. the following prints `worked`.

```
#!/usr/bin/env bash

set -euo pipefail

if ! false; then
    echo "worked"
fi
```

Reply:
Perfect :+1: